### ISSUE TITLE
defaults can need to be reset

### ISSUE THREAD

**Issue by JonathanDoughty** [ForkIssues/Tracker#1601]
No longer an issue for me now, but Fork 2.17 and 2.18 were exiting repeatedly.  tl;dr: the fix was found in [another issue](https://github.com/fork-dev/Tracker/issues/1542): `defaults delete com.DanPristupov.Fork`

Background in case it is relevant:

I've been using Fork for years on multiple macs, but ended up having to restore from a Time Machine backup onto a new arm64 mac when my previous x64 mac got fried. After I did that - and after I had also installed a freshly downloaded copy of Fork (I did not restore any applications) - each time I asked Fork to open a repository it would immediately exit. Attempts to have it re-scan for repositories, with or without keeping the existing structure - or other modifications to Preferences - did not correct the problem and would also cause an immediate exit. Things I tried:

1. Using [AppCleaner](https://freemacsoft.net/appcleaner/) to remove what I thought was all of Fork's bits and pieces before trying fresh installs of both 2.17 and 2.18.

1. Checking the System Preferences ... Security & Privacy... Privacy... Files and Folders permissions for Fork as others have.

1. Looking at the contents of in `~/Library/Logs/Fork.log` but nothing useful to this situation was apparent there.

1. Finally I tried the `defaults delete com.DanPristupov.Fork` mentioned in [#1542](https://github.com/fork-dev/Tracker/issues/1542). That fixed it for me; Fork again works as expected.

I suspect something is initialized or some code path is traversed when defaults are first created that is never re-visited even when a new version is first used. Possibly this only happens when the even more rare underlying machine architecture changes. Forcing Fork to re-initialize those preferences works around the bug.

**Comment by DanPristupov** [comment]
I think you switched Fork to another git instance which became available after the restore.

**Comment by JonathanDoughty** [comment]
True. The system initially had the native macos git and I also installed homebrew's more recent version at some later time. An oddity I encountered when looking at Fork's Preferences during this process was that the git instance selection menu had, literally as I recall, "Menu item 1", "Menu item 2", "Menu item 3", with no indication of which one Fork would use and no custom git instance possibility as it has now.

**Comment by DanPristupov** [comment]
For my reference: https://github.com/fork-dev/Tracker/issues/1488

**Comment by JonathanDoughty** [comment]
My Crash Reports are similar to that in the reference: Fork crashes on an abort() called.

In my case, after I had restored from a backup, I proceeded to install homebrew. Noting the [discussion](https://apple.stackexchange.com/a/437622/57102) about the change to using `/opt/homebrew` I elected to move my restored `/usr/local/bin` aside and fully adopt `/opt/homebrew` for all its installations, as well as cleaning up the other, now outdated, leftovers in `/usr/local`. When I installed Fork there was no `/usr/local/bin` although there was a `/usr/local`. `/usr/local/bin` on my system _now_ is devoted to other, non-homebrew contents, including Fork's own command line integration link.
